### DISCUSSION TITLE
how to use async/await on computed fields

### DISCUSSION THREAD

**Discussion by delkawy** [prisma/prisma#21311] [Q&A]
### Question

Hello, I'm trying to return an async value in a computed field, but the return is always an empty object and not the desired value. Is there any solution or alternative?

```js
export const prisma = new PrismaClient().$extends({
  result: {
    advert: {
      events: {
        async compute(advert) {
          const result = await redis.recover(advert.id);

          return result;
        }
      }
    }
  }
});
```

**Comment by nurul3101** [comment]
Hello @delkawy 👋 

Here's a related GitHub Issue similar to what you are trying to achieve:

- https://github.com/prisma/prisma/issues/18300

I believe this is expected behaviour.
Reference: [comment](https://github.com/prisma/prisma/issues/18300#issuecomment-1468303756).

The GitHub Issue also demonstrates a way through which you could use async compute.

Can you have a look?

**Comment by lacherogwu** [comment]
I would create a view with a join and access the view instead of using async computed properties. This approach is more efficient because if you use `async compute()`, the SQL would not be optimized as it's running on JavaScript, triggering a separate query for each returned row.
